students/ganyoa/lessons/lesson09/assignment/database.py

```diff
@@ -0,0 +1,177 @@
+'''
+Advanced Programming in Python Lesson 9
+Advance Language Constructs; logger context manager
+
+pylint disabled warnings:
+'''
+import csv
+import logging
+import os
+from pymongo import MongoClient
+from pymongo import errors as mongoerr
+
+
+LOG_FORMAT = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+FORMATTER = logging.Formatter(LOG_FORMAT)
+
+FILE_HANDLER = logging.FileHandler('db.log')
+FILE_HANDLER.setLevel(logging.INFO)
+FILE_HANDLER.setFormatter(FORMATTER)
+
+CONSOLE_HANDLER = logging.StreamHandler()
+CONSOLE_HANDLER.setLevel(logging.INFO)
+CONSOLE_HANDLER.setFormatter(FORMATTER)
+
+
+LOGGER = logging.getLogger()
+LOGGER.setLevel(logging.INFO)
+LOGGER.addHandler(FILE_HANDLER)
+LOGGER.addHandler(CONSOLE_HANDLER)
+
+
+class MongoDBConnection():
+    """MongoDB Connection"""
+    def __init__(self, host='127.0.0.1', port=27017):
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+class db_context():
+    '''context manager for repetition'''
```

Comment:
This was interesting. _MongoDBConnection()_ was already a context manager, but it does answer the question: "Can you have a context manager within a context manager?" :-)